feat(codegen): print `vite` / `webpack` special comments

Dunqing opened the pull request:
Related: https://github.com/oxc-project/oxc/issues/1046#issuecomment-2347345319
Close: #6024

Dunqing commented:
* **#6021** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6021?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Dunqing commented on crates/oxc_codegen/tests/integration/esbuild.rs:
```diff
@@ -234,6 +235,37 @@ fn test_call() {
     // testMangleMinify(t, "(1 ? eval : 2)(x)", "(0,eval)(x);");
     // testMangleMinify(t, "(1 ? eval : 2)?.(x)", "eval?.(x);");
+
+    // Webpack-specific comments
+    // TODO: Not support trailing comments yet
+    // test(
+    //     "import(// webpackFoo: 1\n // webpackBar: 2\n 'path');",
+    //     "import(\n  // webpackFoo: 1\n  // webpackBar: 2\n  \"path\"\n);\n",
+    // );
```
Currently, We cannot support this case, because we haven't touch trailing comments

```ts
import(// webpackFoo: 1
 // webpackBar: 2
'path');
```

Dunqing commented:
A lot of `minifier` cases failed, it looks like the compressor caused some cases to assign an incorrect `span`

Dunqing commented on crates/oxc_codegen/tests/integration/snapshots/pure_comments.snap:
```diff
@@ -216,7 +216,9 @@ const builtInSymbols = new Set(
         
 ----------
-const builtInSymbols = new Set(/*#__PURE__*/ Object.getOwnPropertyNames(Symbol).filter((key) => key !== 'arguments' && key !== 'caller'));
+const builtInSymbols = new Set(
+	/*#__PURE__*/ Object.getOwnPropertyNames(Symbol).filter((key) => key !== 'arguments' && key !== 'caller')
+);
```
Same as esbuild https://esbuild.github.io/try/#dAAwLjI0LjAAAGNvbnN0IGJ1aWx0SW5TeW1ib2xzID0gbmV3IFNldCgKICAvKiNfX1BVUkVfXyovCiAgT2JqZWN0LmdldE93blByb3BlcnR5TmFtZXMoU3ltYm9sKQogICAgLmZpbHRlcihrZXkgPT4ga2V5ICE9PSAiYXJndW1lbnRzIiAmJiBrZXkgIT09ICJjYWxsZXIiKQop

Boshen reviewed (changes requested):
Can we add some integration tests for vite special comments, including tests that don't match these webpack / vite patterns?

You may add them to crates/oxc_codegen/tests/integration/unit.rs or merge them in pure_comments.rs

Dunqing commented:
> Can we add some integration tests for vite special comments, including tests that don't match these webpack / vite patterns?
> 
> You may add them to crates/oxc_codegen/tests/integration/unit.rs or merge them in pure_comments.rs

I added tests for vite special comments in crates/oxc_codegen/tests/integration/unit.rs and non-special comments tests which copy from esbuild in esbuild

Dunqing committed:
feat(codegen): print `vite` / `webpack` special comments (#6021)

Related: https://github.com/oxc-project/oxc/issues/1046#issuecomment-2347345319
Close: #6024
